### PR TITLE
Refactor API to use scmUri

### PR DESCRIPTION
**Pull request by stjohnjohnson**
This is so @tkyi can continue working on moving to scmUri

### CODE REVIEW

**Comment by tkyi** [comment]
Need to migrate all the tables to use the new data-schema first

**Comment by minz1027** [comment]
Fixed the functional test to create pipelines dynamically!

File: features/step_definitions/authorization.js [hunk 1]
```diff
@@ -38,7 +33,7 @@ module.exports = function server() {
 
             switch (user) {
             case 'calvin':
-                Assert.strictEqual(decodedToken.username, 'sd-buildbot');
+                Assert.strictEqual(decodedToken.username, 'minz1027');
```
Comment (d2lam) [inline features/step_definitions/authorization.js]: 'sd-buildbot'

**Comment by tkyi** [comment]
Cloned prod `pipeline` and `secret` tables to `rc2_`

File: features/step_definitions/git-flow.js [hunk 2]
```diff
@@ -200,4 +186,22 @@ module.exports = function server() {
             Assert.oneOf(data.state, ['success', 'pending']);
         })
     );
+
+    // eslint-disable-next-line new-cap
+    this.After({
```
Comment (FenrirUnbound) [inline features/step_definitions/git-flow.js]: Just keep in mind that the `.After` block is not guaranteed

Comment (minz1027) [inline features/step_definitions/git-flow.js]: Cannot figure out a better way to do it. :(  We don't know the pipelineId before we create it in the `.Before` hook

Comment (FenrirUnbound) [inline features/step_definitions/git-flow.js]: Do your tests care whether a pipeline was recently created or already exists? Can they tell the difference?

Comment (tkyi) [inline features/step_definitions/git-flow.js]: Our tests don't care, but we need the `pipelineId`. The alternative (to creating a pipeline every time) would be searching through a list of pipelines for a particular `scmUri` after making a call to the scm to get a `repoId` to construct an `scmUri`.

**Pull request merged**